### DISCUSSION TITLE
White screen issue on high screen resolution machine for images of size more than 500 MB

### DISCUSSION THREAD

**Discussion by bhosachin** [cefsharp/CefSharp#4508] [Q&A]
We are using CefSharp.Wpf.HwndHost to launch image viewer application which can render high resolution images having size more than 500 MB. If we use viewer to launch multiple images and try to zoom in and zoom out images then after some time it gets white screen.

![image](https://github.com/cefsharp/CefSharp/assets/107455705/e26080d0-785c-486a-b861-b5dd93e033af)

It works on Chrome but not inside CefSharp.Wpf.HwndHost.

 Screen resolution is 4K and 3840x2160.  Graphics cards from NVIDIA or AMD.

**Debug.log file output** : 

[0301/161320.236:INFO:CONSOLE(129)] "Uncaught RangeError: Failed to execute 'getImageData' on 'CanvasRenderingContext2D': Out of memory at ImageData creation", 

"You are running the esm-bundler build of vue-i18n. It is recommended to configure your bundler to explicitly replace feature flag globals with boolean literals to get proper tree-shaking in the final bundle.",

**Cefsharp setting that we tried** :
            settings.CefCommandLineArgs.Add("enable-speech-input", "1");
            settings.CefCommandLineArgs.Add("allow-universal-access-from-files", "1");
            settings.CefCommandLineArgs.Add("allow-file-access-from-files", "1");
            settings.CefCommandLineArgs.Add("enable-media-stream", "1");
            settings.CefCommandLineArgs.Add("use-fake-ui-for-media-stream", "1");
            settings.CefCommandLineArgs.Add("off-screen-rendering-enabled");
            settings.CefCommandLineArgs.Add("off-screen-frame-rate", "60");
            settings.CefCommandLineArgs.Add("disable-gpu");
            settings.CefCommandLineArgs.Add("disable-gpu-compositing");
            //settings.CefCommandLineArgs.Add("disable-gpu-vsync");
            settings.CefCommandLineArgs.Add("enable-begin-frame-scheduling");
            settings.CefCommandLineArgs.Add("enable-media-stream");

**Comment by amaitland** [comment]
> then after some time it gets white screen.

This is a sign that your `render process` crashed. [OnRenderProcessTerminated](https://cefsharp.github.io/api/111.2.x/html/M_CefSharp_Handler_RequestHandler_OnRenderProcessTerminated.htm) should let you know there was a crash.

> Out of memory at ImageData creation",

Process ran out of memory. Make sure you are using `x64` process.

> It works on Chrome

Are you comparing like for like? Same version, command line ags, bitness, same render process lifespan etc.

Are you running `Chrome` with `GPU disabled`? Saying it works in chrome when you are using different configurations isn't really meaningful.

> settings.CefCommandLineArgs.Add("off-screen-rendering-enabled");
> settings.CefCommandLineArgs.Add("off-screen-frame-rate", "60");

Not relevant, remove these.

> settings.CefCommandLineArgs.Add("disable-gpu");
> settings.CefCommandLineArgs.Add("disable-gpu-compositing");

GPU should be left enabled, make sure your app has an `app.manifest` with compatability entries (refer to the quick start)

> settings.CefCommandLineArgs.Add("enable-begin-frame-scheduling");

Remove this, shouldn't be used with `HwndHost` implementation.

Make sure you are using the current stable version.
